aries_cloudagent/protocols/didexchange/v1_0/manager.py

```diff
@@ -474,23 +497,28 @@ async def receive_request(
                 conn_rec = new_conn_rec
 
         # request DID doc describes requester DID
-        if not (request.did_doc_attach and request.did_doc_attach.data):
-            raise DIDXManagerError(
-                "DID Doc attachment missing or has no data: "
-                "cannot connect to public DID"
-            )
-        async with self.profile.session() as session:
-            wallet = session.inject(BaseWallet)
-            conn_did_doc = await self.verify_diddoc(wallet, request.did_doc_attach)
-        if request.did != conn_did_doc.did:
-            raise DIDXManagerError(
-                (
-                    f"Connection DID {request.did} does not match "
-                    f"DID Doc id {conn_did_doc.did}"
-                ),
-                error_code=ProblemReportReason.REQUEST_NOT_ACCEPTED.value,
+        if request.did_doc_attach and request.did_doc_attach.data:
+            self._logger.debug("Received DID Doc attachment in request")
+            async with self.profile.session() as session:
+                wallet = session.inject(BaseWallet)
+                conn_did_doc = await self.verify_diddoc(wallet, request.did_doc_attach)
```

Comment:
What is the DIDDoc deserialize throws an error? the DIDDoc class looks for `publickey` instead of `verificationMethods`. 

https://github.com/dbluhm/aries-cloudagent-python/blob/4de1a12586456feef1fed231610f1391ba96bf99/aries_cloudagent/connections/models/diddoc/diddoc.py#L216

Do we need to check the context attribute of the diddoc before deserializing?

Reply:
Is this verifying a did_doc_attach from the other agent? or one resolved from a public ledger?

Reply:
This is unchanged from the original handling of DID Document attachments. I would consider any changes here to be out of scope for this PR. For changes to this behavior beyond this PR, I think we effectively side step it altogether by using Peer DIDs (we won't include an attached DID Doc for peer DIDs).